src/twisted/plugins/cred_unix.py

```diff
@@ -65,7 +68,14 @@ def checkPwd(self, pwd, username, password):
 
     def checkSpwd(self, spwd, username, password):
         try:
-            cryptedPass = spwd.getspnam(username)[1]
+            if not isinstance(username, StringType):
```

Comment:
This comment is preference, meaning I'm not advocating for it, but I wanted to point it out for consideration.

In nearly all cases, I dislike magic numbers for an index.  This is one of the rare cases where I actually prefer it.  My reasoning is that because the indexes are explicitly described in the standard library docs they provide a succinct way to access the encrypted password without caring about the python version.  Unless subsequent viewers of this code are familiar with getspnam() then it's likely they'll go to the std library docs anyway.  Were the code to use the old indexing I would suggest a comment describing where the index is documented.

Reply:
I'm going to disagree with you on that one.  If someone has to look into the standard library,
these fields are documented quite visibly:

https://docs.python.org/2/library/pwd.html
https://docs.python.org/2/library/spwd.html

and Python 3:

https://docs.python.org/3/library/pwd.html
https://docs.python.org/3/library/spwd.html

Having the field names spelled out is more readable than some index offset.
If someone needs to look up the field, it is quite well documented.

Reply:
No worries, like I said, this was opinion on my part.